src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -537,70 +547,136 @@ public static class SystemNamespaceType
                     }
                 }
 
-                return LanguageConstants.String;
+                return new(LanguageConstants.String);
             };
 
-        private static TypeSymbol LoadTextContentTypeBuilder(IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, ImmutableArray<FunctionArgumentSyntax> arguments, ImmutableArray<TypeSymbol> argumentTypes)
+        private static FunctionResult LoadTextContentResultBuilder(IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, ImmutableArray<FunctionArgumentSyntax> arguments, ImmutableArray<TypeSymbol> argumentTypes)
         {
-            if (argumentTypes[0] is not StringLiteralType filePathType)
+            return TryLoadTextContentFromFile(binder, fileResolver, diagnostics,
+                (arguments[0], argumentTypes[0]),
+                arguments.Length > 1 ? (arguments[1], argumentTypes[1]) : null,
+                out var fileContent,
+                LanguageConstants.MaxLiteralCharacterLimit)
+                ? new(new StringLiteralType(fileContent))
+                : new(LanguageConstants.String);
+        }
+
+        private static FunctionResult LoadJsonContentResultBuilder(IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, ImmutableArray<FunctionArgumentSyntax> arguments, ImmutableArray<TypeSymbol> argumentTypes)
+        {
+            string? tokenSelectorPath = null;
+            if (arguments.Length > 1)
             {
-                diagnostics.Write(DiagnosticBuilder.ForPosition(arguments[0]).CompileTimeConstantRequired());
-                return LanguageConstants.String;
+                if (argumentTypes[1] is not StringLiteralType tokenSelectorType)
+                {
+                    diagnostics.Write(DiagnosticBuilder.ForPosition(arguments[1]).CompileTimeConstantRequired());
+                    return new(LanguageConstants.Any);
```

Comment:
On lines 589-590 you're returning an `ErrorType` with diagnostics attached. I think it would be good to be consistent and follow that same approach here - that way the type system will know the type is 'error' rather than a valid type.

Reply:
it's probably a copy-paste leftover as I was reusing I think yours code :) as I wrote in other comment - I don't entirely get the ErrorType and when I should use it rather than combination of diagnostics.Error + default function return type.

Reply:
It's definitely a minor point, but IMO better to be accurate. The main thing is that it has an impact on downstream type analysis. As an example, imagine the following scenario:

```bicep
var foo = loadJsonContent('nonexistentfile.txt')
var bar = foo + 1
```

With your implementation, if you hover over `foo` and `bar`, you'll see types as `any` and `int` respectively.
If you propagate the error type properly, the type for both should be shown as `error`

Reply:
good point, I'll change to error type

Reply:
changed